From 51d2a8114b94193ab5bf78a9bff7b6b0a0b7a07c Mon Sep 17 00:00:00 2001
From: Roman Li <Roman.Li@amd.com>
Date: Wed, 17 Aug 2016 15:51:52 -0400
Subject: [PATCH 0826/1722] drm/amd/dal: add data entries for virtual signal
 type

This fixes link training failure for EDID emulation usecases

Signed-off-by: Roman Li <Roman.Li@amd.com>
Signed-off-by: Andrey Grodzovsky <Andrey.Grodzovsky@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c  | 1 +
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_clock_source.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c b/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
index a27db8c..1fe7f9b 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table_helper.c
@@ -160,6 +160,7 @@ uint32_t dal_cmd_table_helper_encoder_mode_bp_to_atom(
 	case SIGNAL_TYPE_EDP:
 	case SIGNAL_TYPE_DISPLAY_PORT_MST:
 	case SIGNAL_TYPE_DISPLAY_PORT:
+	case SIGNAL_TYPE_VIRTUAL:
 		if (enable_dp_audio)
 			return ATOM_ENCODER_MODE_DP_AUDIO;
 		else
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_clock_source.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_clock_source.c
index 5e315ba..3509c5d4 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_clock_source.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_clock_source.c
@@ -67,6 +67,7 @@ static const struct spread_spectrum_data *get_ss_data_entry(
 	case SIGNAL_TYPE_DISPLAY_PORT:
 	case SIGNAL_TYPE_DISPLAY_PORT_MST:
 	case SIGNAL_TYPE_EDP:
+	case SIGNAL_TYPE_VIRTUAL:
 		ss_parm = clk_src->dp_ss_params;
 		entrys_num = clk_src->dp_ss_params_cnt;
 		break;
-- 
2.7.4

